### PR TITLE
Strict mode: strictNullChecks part 3 (structures/helpers)

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/command_prechecks.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default class CommandPrechecks {
             messageContext.guildID
         );
 
-        if (session.isListeningSession()) {
+        if (session.isListeningSession() && interaction) {
```
Comment (taahamahdi) [inline src/command_prechecks.ts:50]: What about an in session check triggered by a non-interaction command, with the bot and user in different channels? We'd skip entering here and would be casting the ListeningSession as a GameSession

File: src/events/client/interactionCreate.ts [hunk 2]
```diff
@@ -104,6 +104,7 @@ export default async function interactionCreateHandler(
                 logger.error(
                     `No handler found for CHAT_INPUT CommandInteraction: ${interaction.data.name}`
                 );
+                return;
```
Comment (taahamahdi) [inline src/events/client/interactionCreate.ts:107]: Are the early returns needed for `No handler found` cases? We already return if `interactionPromise === null`

File: src/helpers/validate.ts [hunk 3]
```diff
@@ -94,7 +94,7 @@ export default (
 
                 // parse as integer for now, might cause problems later?
                 const intArg = parseInt(arg, 10);
-                if ("minValue" in validation && intArg < validation.minValue) {
+                if (validation.minValue && intArg < validation.minValue) {
```
Comment (taahamahdi) [inline src/helpers/validate.ts:97]: If `minValue` is zero this check never continues

File: src/helpers/validate.ts [hunk 4]
```diff
@@ -111,7 +111,7 @@ export default (
                     return false;
                 }
 
-                if ("maxValue" in validation && intArg > validation.maxValue) {
+                if (validation.maxValue && intArg > validation.maxValue) {
```
Comment (taahamahdi) [inline src/helpers/validate.ts:114]: If `maxValue` is zero this check never continues (less likely but still possible)

File: src/kmq.ts [hunk 5]
```diff
@@ -180,15 +201,18 @@ async function startWebServer(fleet: Fleet): Promise<void> {
             return;
         }
 
-        const isSoftRestart = request.body["soft"] ? 1 : 0;
+        const isSoftRestart = (request.body as any)["soft"] ? 1 : 0;
```
Comment (taahamahdi) [inline src/kmq.ts:204]: Nit: Can cast to an object with `soft` property

File: src/kmq.ts [hunk 6]
```diff
@@ -180,15 +201,18 @@ async function startWebServer(fleet: Fleet): Promise<void> {
...
 
         if (isSoftRestart) {
             logger.info("Soft restart initiated");
             await exec("npx tsc");
             fleet.restartAllClusters(false);
         }
 
-        const restartMinutes = request.body["restartMinutes"];
+        const restartMinutes = (request.body as any)[
```
Comment (taahamahdi) [inline src/kmq.ts:212]: Nit: Can cast to an object with `restartMinutes` property

File: src/kmq.ts [hunk 7]
```diff
@@ -216,7 +240,7 @@ async function startWebServer(fleet: Fleet): Promise<void> {
             return;
         }
 
-        const userID = request.body["user"];
+        const userID = (request.body as any)["user"] as string;
```
Comment (taahamahdi) [inline src/kmq.ts:243]: Nit: Can cast to an object with `user` property

File: src/seed/bootstrap.ts [hunk 8]
```diff
@@ -69,8 +69,8 @@ async function songThresholdReached(db: DatabaseContext): Promise<boolean> {
     if (!availableSongsTableExists) return false;
 
     return (
-        (await db.kmq("available_songs").count("* as count").first()).count >=
-        SONG_DOWNLOAD_THRESHOLD
+        ((await db.kmq("available_songs").count("* as count").first()) as any)
```
Comment (taahamahdi) [inline src/seed/bootstrap.ts:72]: Nit: Can cast to an object with `count` property

File: src/seed/seed_db.ts [hunk 9]
```diff
@@ -63,12 +63,12 @@ export async function tableExists(
...
     return (
         (
-            await db
+            (await db
                 .agnostic("information_schema.tables")
                 .where("table_schema", "=", databaseName)
                 .where("table_name", "=", tableName)
                 .count("* as count")
-                .first()
+                .first()) as any
```
Comment (taahamahdi) [inline src/seed/seed_db.ts:71]: Nit: Can cast to an object with `count` property

File: src/seed/seed_db.ts [hunk 10]
```diff
@@ -247,32 +247,32 @@ async function validateSqlDump(
 
         logger.info("Validating MV song count");
         const mvSongCount = (
-            await db
+            (await db
                 .kpopVideosValidation("app_kpop")
                 .count("* as count")
                 .where("is_audio", "=", "n")
-                .first()
+                .first()) as any
```
Comment (taahamahdi) [inline src/seed/seed_db.ts:254]: Nit: Can cast to an object with `count` property

File: src/seed/seed_db.ts [hunk 11]
```diff
@@ -247,32 +247,32 @@ async function validateSqlDump(
...
 
         logger.info("Validating audio-only song count");
         const audioSongCount = (
-            await db
+            (await db
                 .kpopVideosValidation("app_kpop")
                 .count("* as count")
                 .where("is_audio", "=", "y")
-                .first()
+                .first()) as any
```
Comment (taahamahdi) [inline src/seed/seed_db.ts:265]: Nit: Can cast to an object with `count` property

File: src/seed/seed_db.ts [hunk 12]
```diff
@@ -247,32 +247,32 @@ async function validateSqlDump(
...
         logger.info(`Found ${audioSongCount} audio-only videos`);
 
         logger.info("Validating group count");
         const artistCount = (
-            await db
+            (await db
                 .kpopVideosValidation("app_kpop_group")
                 .count("* as count")
-                .first()
+                .first()) as any
```
Comment (taahamahdi) [inline src/seed/seed_db.ts:275]: Nit: Can cast to an object with `count` property

File: src/structures/listening_session.ts [hunk 13]
```diff
@@ -156,11 +154,11 @@ export default class ListeningSession extends Session {
         }
 
         const round = this.round;
-        const guildID = interaction.guildID;
+        const guildID = interaction.guildID as string;
         if (interaction.data.custom_id === "bookmark") {
             this.handleBookmarkInteraction(interaction);
```
Comment (taahamahdi) [inline src/structures/listening_session.ts:159]: Do we need to await this?

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/command_prechecks.ts (hunk 1 above)
Comment (Brainicism) [reply src/command_prechecks.ts:50]: This conditional is only for interaction response via `tryCreateInteractionErrorAcknowledgement`. If we want to make it the same behavior for non-interactions then we need another PR for it.

File: src/structures/listening_session.ts (hunk 13 above)
Comment (Brainicism) [reply src/structures/listening_session.ts:159]: Don't know but not within the scope of this PR

Comment (taahamahdi) [reply src/structures/listening_session.ts:159]: This PR makes `handleBookmarkInteraction` async so we might need to await where appropriate

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**